File: jscomp/test/inline_edge_cases.js
'use strict';


function test3(_n) {
  while(true) {
    var n = _n;
    if (n === 0) {
      return (n + 5 | 0) + 4 | 0;
    } else {
      _n = n - 1 | 0;
      continue ;
      
    }
  };
}

function test2(_n) {
  while(true) {
    var n = _n;
    if (n === 0) {
      return test3(n) + 3 | 0;
    } else {
      _n = n - 1 | 0;
      continue ;
      
    }
  };
}

function test0(_n) {
  while(true) {
    var n = _n;
    if (n === 0) {
      var _n$1 = n;
      while(true) {
        var n$1 = _n$1;
        if (n$1 === 0) {
          return test2(n$1) + 2 | 0;
        } else {
          _n$1 = n$1 - 1 | 0;
          continue ;
          
        }
      };
    } else {
      _n = n - 1 | 0;
      continue ;
      
    }
  };
}

var v = test0(10);

test0(10) + 2 | 0;

exports.v = v;
/* v Not a pure module */

File: jscomp/test/inline_edge_cases.ml
let test4 n = n + 5
let rec test3 n = if n = 0 then test4 n + 4 else test3 (n - 1)
let rec test2 n = if n = 0 then test3 n + 3 else test2 (n - 1)
let rec test1 n = if n = 0 then test2 n + 2 else test1 (n - 1)
let rec test0 n = if n = 0 then test1 n else test0 (n - 1)
let v = test0 10
let u = test0 10 + 2
